In [0]:
%sql
USE CATALOG conecta;
USE SCHEMA alarmes_aju;

## 1. Número de alarmes por altura de poste.

Os pontos de serviço (postes) com altura superior ou igual a 13 metros, são considerados "altos", pois os caminhões que a empresa tem disponíveis alcançam uma altura máxima inferior a 13 metros.
Essa pergunta serve para que o gerente possa saber a diferença entre alarmes que são em postes baixos e altos. Além disso, é necessário filtrar apenas os alarmes Sem Comunicação e Falha no Driver, pois esses são os que impactam necessariamente essa análise.

Na consulta é utilizado SELECT DISTINCT com IDPontoServico, que é a chave natural de PontoServico. Isso significa que só estamos pegando um registro por poste. Caso um poste possua mais que uma LCU com os mesmos tipos de alarme, será retornado apenas um registro. Isso é interessante, pois postes alto tendem a possuir 4 LCUs, o que iria inflar muito o resultado a favor dos postes altos.

In [0]:
%sql
CREATE OR REPLACE VIEW viewAlarmePontoServico AS (
  SELECT NomeAlarme, TipoPoste, IDPontoServico, Data, TrimestreContrato, Bairro
  FROM fatoalarme
  LEFT JOIN dimtipoalarme ON fatoalarme.AlarmeKey = dimtipoalarme.AlarmeKey
  LEFT JOIN dimpontoservico ON fatoalarme.PontoServicoKey = dimpontoservico.PontoServicoKey
  LEFT JOIN dimdata ON fatoalarme.DataKey = dimdata.DataKey
  WHERE NomeAlarme IN ('Sem Comunicação', 'Falha no Driver')
);

In [0]:
%sql
SELECT NomeAlarme, TipoPoste, count(DISTINCT IDPontoServico) NumeroPostes
FROM viewAlarmePontoServico
WHERE data = (SELECT MAX(Data) FROM dimdata)
GROUP BY NomeAlarme, TipoPoste
ORDER BY TipoPoste;

NomeAlarme TipoPoste NumeroPostes Sem Comunicação Poste Alto 20 Falha no Driver Poste Alto 22 Sem Comunicação Poste Baixo 29 Falha no Driver Poste Baixo 7

### Discussão
O número de alarmes em pontos de serviço baixo é semelhante ao de poste alto. Isso indica que a situação atual é de que temos aproximadamente o mesmo número de ocorrências de poste baixo e poste alto, evidenciando a necessidade de alugar o caminhão de alcance maior para dar manutenção nos pontos de poste alto.

Sabemos que provavelmente o número de LCUs em poste alto será maior, pois como informado no enunciado, postes altos tendem a possuir 4 LCUs, existindo a possibilidade de que mais do que uma LCU possua alarmes.

## 2. Número de Alarmes no trimestre do contrato atual por altura do poste.

Da mesma forma que na pergunta anterior, estamos interessados no número de alarmes em IDs Ponto Servico distintos, então será utilizado a visão viewAlarmePontoServico como base da consulta.

In [0]:
%sql
SELECT NomeAlarme, TipoPoste, count(DISTINCT IDPontoServico) NumeroPostes
FROM viewAlarmePontoServico
WHERE TrimestreContrato IN (
  SELECT TrimestreContrato
  FROM viewAlarmePontoServico
  WHERE Data = (SELECT MAX(Data) FROM dimdata)
)
GROUP BY NomeAlarme, TipoPoste
ORDER BY TipoPoste;

NomeAlarme TipoPoste NumeroPostes Sem Comunicação Poste Alto 57 Falha no Driver Poste Alto 33 Sem Comunicação Poste Baixo 194 Falha no Driver Poste Baixo 29

### Discussão
Claramente a maior proporção de alarmes no trimestre atual é de postes baixos. Isso revela uma tendência de que a maior parte dos serviços são para postes baixos.

A proporção de postes altos em relação ao total de postes com sistema de telegestão é de 17 %. O resultado dessa consulta mostra que 29 % dos alarmes no trimestre foram em postes altos, revelando um aumento proporcional de ocorrências em postes altos.

## 3. Número de alarmes no último ETL

É interessante saber o número de alarmes do fechamento do dia anterior - o último ETL.

In [0]:
%sql
CREATE OR REPLACE VIEW viewAlarmeLCU AS (
  SELECT NomeLCU, NomeAlarme, Data, TrimestreContrato, IDLCU, Latitude, Longitude
  FROM fatoalarme a
  LEFT JOIN dimlcu ON dimlcu.LCUKey = a.LCUKey
  LEFT JOIN dimdata ON dimdata.DataKey = a.DataKey
  LEFT JOIN dimtipoalarme ON dimtipoalarme.AlarmeKey = a.AlarmeKey
)

In [0]:
%sql
SELECT NomeAlarme, count(NomeAlarme) NumeroAlarmes
FROM viewalarmelcu
WHERE Data = (SELECT MAX(Data) FROM dimdata)
GROUP BY NomeAlarme
ORDER BY NumeroAlarmes DESC;

NomeAlarme NumeroAlarmes Sem Comunicação 53 Acesa Durante o Dia 49 Falha no Driver 31 Apagada Durante a Noite 29 Tensão Alta 10 Tensão Baixa 5

### Discussão
A consulta serve apenas para evidenciar o número de alarmes por Nome de Alarme na data de última carga de dados.

## 4. Número de alarmes resolvidos no trimestre do contrato.

Foi utilizada a heurística de que se um alarme não surge novamente em dada LCU, em um período arbitrário, significa que o defeito foi resolvido.
Exemplo:
LCU 1000 no dia 10/10/2025 apresenta alarme Falha no Driver. Se no dia 11/10/2025 e 12/10/2025 não apareceu mais alarmes de Falha no Driver na LCU 1000, será considerado que o problema foi resolvido.
Essa margem é utilizada, pois existem alarmes que surgem em intervalos curtos de tempo por instabilidade na rede LoRaWan ou rede de energia. Além disso, alarmes realmente resolvidos tendem a não reaparecer por bastante tempo.

O seguinte código originalmente respondia a pergunta 5. Minha resposta da pergunta 5 era muito mal otimizada (código disponível na pergunta 5) e utilizei o Gemini para me ajudar a criar uma consulta baseada somente em SQL. Percebi que com algumas pequenas modificações eu poderia criar uma View para responder a pergunta 4 e 5. Também aprendi a lógica aplicada por esse SQL do Gemini e consegui criar consultas até mais complexas nas próximas perguntas, sem ajuda.

In [0]:
%sql
CREATE OR REPLACE VIEW viewAlarmesResolvidos AS (
  SELECT v1.*
  FROM viewalarmelcu v1
  WHERE v1.Data <= date_sub((SELECT MAX(Data) FROM dimdata), 2)
  -- Só consigo ter certeza que um alarme está resolvido tendo dados de dois dias posteriores à análise.
  AND TrimestreContrato IN (
    SELECT TrimestreContrato
    FROM viewalarmelcu
    WHERE Data = (SELECT MAX(Data) FROM dimdata)
  )
  AND NOT EXISTS (
    SELECT 1
    FROM viewalarmelcu v2
    WHERE v1.NomeAlarme = v2.NomeAlarme
    AND v1.IDLCU = V2.IDLCU
    AND v2.Data > v1.Data
    AND v2.Data <= date_add(v1.Data, 2)
  )
)

In [0]:
%sql
SELECT NomeAlarme, count(NomeAlarme) NumeroAlarmesResolvidos
FROM viewAlarmesResolvidos
GROUP BY NomeAlarme
ORDER BY NumeroAlarmesResolvidos DESC;

NomeAlarme NumeroAlarmesResolvidos Sem Comunicação 258 Acesa Durante o Dia 182 Apagada Durante a Noite 73 Falha no Driver 45 Tensão Baixa 3

### 4.2 LCUs e Alarmes distintos resolvidos no trimestre do contrato.

In [0]:
%sql
SELECT NomeAlarme, count(DISTINCT NomeAlarme, NomeLCU) NumeroAlarmesResolvidos
FROM viewAlarmesResolvidos
GROUP BY NomeAlarme
ORDER BY NumeroAlarmesResolvidos DESC;

NomeAlarme NumeroAlarmesResolvidos Sem Comunicação 247 Acesa Durante o Dia 146 Apagada Durante a Noite 54 Falha no Driver 37 Tensão Baixa 3

### Discussão
O número elevado de alarmes de Sem Comunicação solucionados é uma consequência direta da atmosfera da cidade de Aracaju que é uma das cidades do Brasil com maior maresia. Esses alarmes ocorrem por conta que o cabo de 1,5 mm² que alimenta as luminária LEDs é partido por causa do salitre, deixando a luminária sem energia e, consequentemente, a LCU fica sem comunicação.

Em segundo lugar aparecem os alarmes de Acesa Durante o Dia. Esses alarmes, na verdade, não foram resolvidos por equipes de manutenção. Ocorre um problema no sistema de telegestão em que as LCUs comandam as luminárias para ficarem acesas durante o dia. Essa noção que o problema é da LCU foi descoberta após a solução da pergunta 8 desse notebook. Até o momento não sabíamos que acontecia esse tipo de situação com as LCUs.

O número de soluções de alarmes Falha no Driver é bem próximo do número de luminárias LED que removemos do campo por estarem queimadas. Isso indica forte relação entre o alarme com um defeito físico da luminária em que a LCU está instalada.

## 5. Número de alarmes resolvidos por dia do Trimestre do Contrato.

Essa pergunta é bem semelhante a anterior e pode ser resolvida pela mesma View, inclusive. Porém, quando a pergunta foi criada eu não tinha noção alguma de como poderia responder a pergunta.

O código a seguir foi minha solução para o problema. Coloquei no Gemini para me trazer uma versão apenas em SQL e ele me trouxe um código SQL como resultado. Aprendi o método do Gemini e vou utilizar no meu trabalho.

A ideia original era utilizar um for loop do python para iterar em todas as datas de DimData. Com a ajuda do Gemini, tive a noção que poderia utilizar tabela virtual para fazer justamente um processo que tem resultado equivalente ao for loop do python, porém muito mais eficiente.

In [0]:
%skip
import pyspark.pandas as ps
dias_trimestre = spark.sql(
    '''
    SELECT DISTINCT Data FROM viewalarmelcu
    WHERE Data < last_dt
    AND TrimestreContrato IN (
        SELECT TrimestreContrato
        FROM viewalarmelcu
        WHERE Data = last_dt
    )
    ''')
dias_trimestre = ps.DataFrame(dias_trimestre)['Data'].sort_values()
result_sql = None
for dia_ in dias_trimestre.to_numpy():
    spk_result = spark.sql(
        f'''
        SELECT Data, Alarme, count(DISTINCT IDLCU, Alarme) as NumeroAlarmesResolvidos
        FROM viewalarmelcu
        WHERE
        (IDLCU, Alarme) NOT IN (
        SELECT IDLCU, Alarme
        FROM viewalarmelcu
        WHERE Data > "{dia_}"
        AND Data <= date_add("{dia_}", 5)
        )
        AND Data = "{dia_}"
        AND Alarme IN ('Sem Comunicação', 'Falha no Driver')
        GROUP BY Data, Alarme
        ORDER BY Data ASC, numeroalarmesresolvidos DESC;
        '''
    )
    if result_sql is None:
        result_sql = spk_result
    else:
        result_sql = result_sql.union(spk_result)
display(result_sql)

In [0]:
%sql
SELECT Data, NomeAlarme, count(IDLCU) AlarmesResolvidos
FROM viewalarmesresolvidos
GROUP BY Data, NomeAlarme
ORDER BY Data, alarmesresolvidos DESC;

Data NomeAlarme AlarmesResolvidos 2025-12-01 Acesa Durante o Dia 16 2025-12-01 Sem Comunicação 11 2025-12-01 Falha no Driver 2 2025-12-01 Apagada Durante a Noite 1 2025-12-02 Sem Comunicação 17 2025-12-02 Acesa Durante o Dia 15 2025-12-02 Apagada Durante a Noite 4 2025-12-02 Falha no Driver 4 2025-12-03 Sem Comunicação 29 2025-12-03 Acesa Durante o Dia 12 2025-12-03 Falha no Driver 5 2025-12-03 Apagada Durante a Noite 2 2025-12-04 Sem Comunicação 13 2025-12-04 Acesa Durante o Dia 10 2025-12-04 Apagada Durante a Noite 4 2025-12-04 Falha no Driver 3 2025-12-04 Tensão Baixa 1 2025-12-05 Sem Comunicação 37 2025-12-05 Apagada Durante a Noite 6 2025-12-05 Acesa Durante o Dia 6 2025-12-05 Falha no Driver 3 2025-12-05 Tensão Baixa 2 2025-12-06 Apagada Durante a Noite 11 2025-12-06 Sem Comunicação 10 2025-12-06 Acesa Durante o Dia 10 2025-12-06 Falha no Driver 4 2025-12-07 Acesa Durante o Dia 15 2025-12-07 Sem Comunicação 7 2025-12-07 Apagada Durante a Noite 6 2025-12-07 Falha no Driver 2 2025-12-08 Sem Comunicação 10 2025-12-08 Acesa Durante o Dia 10 2025-12-08 Falha no Driver 3 2025-12-08 Apagada Durante a Noite 2 2025-12-09 Sem Comunicação 15 2025-12-09 Acesa Durante o Dia 13 2025-12-09 Apagada Durante a Noite 10 2025-12-09 Falha no Driver 5 2025-12-10 Acesa Durante o Dia 18 2025-12-10 Sem Comunicação 15 2025-12-10 Apagada Durante a Noite 7 2025-12-11 Sem Comunicação 16 2025-12-11 Acesa Durante o Dia 12 2025-12-11 Apagada Durante a Noite 5 2025-12-11 Falha no Driver 5 2025-12-12 Sem Comunicação 53 2025-12-12 Acesa Durante o Dia 18 2025-12-12 Apagada Durante a Noite 6 2025-12-12 Falha no Driver 3 2025-12-13 Sem Comunicação 12 2025-12-13 Acesa Durante o Dia 12 2025-12-13 Apagada Durante a Noite 5 2025-12-13 Falha no Driver 2 2025-12-14 Acesa Durante o Dia 15 2025-12-14 Sem Comunicação 13 2025-12-14 Apagada Durante a Noite 4 2025-12-14 Falha no Driver 4

### Discussão
Além do benefício de sabermos quantos alarmes por dia resolvemos, podemos visualizar uma situação de solução de problema em comandos elétricos, indicado pelo alto número de soluções de alarmes Sem Comunicação.
Por exemplo, no dia 12/12, o número de 53 LCUs sem comunicação é explicado, em grande parte, por conta de um comando elétrico com 48 LCUs que ficou fora de serviço. Essas 48 LCUs ficaram sem energia - sem comunicação - e no dia 12/12 resolvemos o problema.

## 6. LCUs com maior frequência de alarmes nos últimos 30 dias.

Será realizada essa consulta considerando apenas os alarmes Sem Comunicação e Falha no Driver que são os de maior interesse para a gerência.

In [0]:
%sql
SELECT NomeLCU, NomeAlarme, count(DISTINCT NomeLCU, Data) NumeroAlarmes, numeroalarmes / 30 * 100 Frequencia
FROM viewalarmelcu
WHERE NomeAlarme IN ('Sem Comunicação', 'Falha no Driver')
AND Data > date_sub((SELECT MAX(Data) FROM dimdata), 30)
GROUP BY NomeLCU, NomeAlarme
ORDER BY NumeroAlarmes DESC
LIMIT 50;

NomeLCU NomeAlarme NumeroAlarmes Frequencia LCU 602338 Sem Comunicação 30 100.0 LCU 6002C7 Sem Comunicação 30 100.0 LCU 1111 Falha no Driver 30 100.0 LCU 1545 Sem Comunicação 30 100.0 LCU 503212 Sem Comunicação 30 100.0 LCU 603325 Sem Comunicação 30 100.0 LCU 604CB6 Falha no Driver 30 100.0 LCU 26EC Sem Comunicação 30 100.0 LCU 602494 Sem Comunicação 30 100.0 LCU 33FF Sem Comunicação 30 100.0 LCU 129B Sem Comunicação 30 100.0 LCU 288E Falha no Driver 30 100.0 LCU 601358 Sem Comunicação 30 100.0 LCU 2469 Sem Comunicação 30 100.0 LCU 602EFB Sem Comunicação 30 100.0 LCU 2727 Sem Comunicação 30 100.0 LCU 604E92 Falha no Driver 30 100.0 LCU 182D Sem Comunicação 30 100.0 LCU 6041A3 Sem Comunicação 30 100.0 LCU 1A10 Falha no Driver 30 100.0 LCU 1EC8 Falha no Driver 30 100.0 LCU 601237 Falha no Driver 30 100.0 LCU 603379 Sem Comunicação 30 100.0 LCU 6042CE Sem Comunicação 30 100.0 LCU 600A38 Sem Comunicação 30 100.0 LCU 602B05 Sem Comunicação 30 100.0 LCU 60241E Sem Comunicação 30 100.0 LCU 60449C Falha no Driver 30 100.0 LCU 603676 Sem Comunicação 30 100.0 LCU 5038E6 Sem Comunicação 30 100.0 LCU 3A6F Sem Comunicação 30 100.0 LCU 1C6E Falha no Driver 30 100.0 LCU 1ED1 Falha no Driver 29 96.66666666666667 LCU 18AD Sem Comunicação 28 93.33333333333333 LCU 601817 Sem Comunicação 28 93.33333333333333 LCU 604666 Sem Comunicação 28 93.33333333333333 LCU 31EA Sem Comunicação 27 90.0 LCU 2F4B Falha no Driver 27 90.0 LCU 1294 Falha no Driver 26 86.66666666666667 LCU 383B Falha no Driver 26 86.66666666666667 LCU 2EF3 Falha no Driver 26 86.66666666666667 LCU 600AA0 Sem Comunicação 25 83.33333333333334 LCU 6054E7 Falha no Driver 25 83.33333333333334 LCU 60235B Falha no Driver 25 83.33333333333334 LCU 1530 Falha no Driver 24 80.0 LCU 1AD9 Sem Comunicação 24 80.0 LCU 14EC Sem Comunicação 24 80.0 LCU 1E26 Sem Comunicação 24 80.0 LCU 6005AB Falha no Driver 24 80.0 LCU 1E61 Sem Comunicação 24 80.0

### Discussão
A consulta mostra quais LCUs estão com maior frequência de alarmes, considerando dados dos últimos 30 dias. As LCUs no topo do resultado devem possuir prioridade para resoluções de problemas por parte da operação.
Esses dados servem para que a gerência cobre uma atitude da operação em relação a esses alarmes.

A maior parte desses alarmes, entretanto, são de situações com impossibilidade. Ou seja, dependemos de atuação de terceiros para que uma providência possa ser tomada.

## 7. LCUs com maior frequência de alarme Aceso Durante o Dia nos últimos 30 dias.

Temos um problema com o alarme de LED Aceso Durante o Dia, pois aparentemente o próprio sistema de telegestão está gerando esse defeito, deixando algumas LEDs da cidade acesas durante o dia. Quando uma equipe faz a intervenção e troca o dispositivo IoT, muitas vezes o dispositivo está apagando normalmente ao excitar o relé, em testes de bancada.
Percebemos que após alguns dias com o defeito a LED em campo apaga e o dispositivo IoT LCU para de gerar alarmes. Talvez, então, vários dias seguidos com esse alarme indique que realmente existe um defeito e não é algo transitório.

In [0]:
%sql
SELECT NomeLCU, count(NomeLCU) NumeroAlarmes, numeroalarmes / 30 * 100 Frequencia
FROM viewalarmelcu
WHERE NomeAlarme = 'Acesa Durante o Dia'
AND Data > date_sub((SELECT MAX(Data) FROM dimdata), 30)
GROUP BY NomeLCU
ORDER BY NumeroAlarmes DESC
LIMIT 50;

NomeLCU NumeroAlarmes Frequencia LCU 2C38 30 100.0 LCU 6008AA 30 100.0 LCU 195D 30 100.0 LCU 13FE 29 96.66666666666667 LCU 1CE5 29 96.66666666666667 LCU 1375 28 93.33333333333333 LCU 2DC8 28 93.33333333333333 LCU 1405 28 93.33333333333333 LCU 1849 28 93.33333333333333 LCU 108A 28 93.33333333333333 LCU 3A54 27 90.0 LCU 12BD 26 86.66666666666667 LCU 3A18 25 83.33333333333334 LCU 177A 25 83.33333333333334 LCU 6053AB 25 83.33333333333334 LCU 1D46 23 76.66666666666667 LCU 189C 22 73.33333333333333 LCU 1F69 22 73.33333333333333 LCU 290C 19 63.33333333333333 LCU 1CFD 19 63.33333333333333 LCU 3758 19 63.33333333333333 LCU 1226 18 60.0 LCU 2143 18 60.0 LCU 602D2B 17 56.666666666666664 LCU 17EA 17 56.666666666666664 LCU 602487 17 56.666666666666664 LCU 2911 17 56.666666666666664 LCU 502108 15 50.0 LCU 3172 14 46.666666666666664 LCU 503963 14 46.666666666666664 LCU 2CB7 14 46.666666666666664 LCU 2692 13 43.333333333333336 LCU 604260 13 43.333333333333336 LCU 1913 13 43.333333333333336 LCU 3AB9 13 43.333333333333336 LCU 2607 13 43.333333333333336 LCU 2E53 11 36.666666666666664 LCU 2E10 11 36.666666666666664 LCU 102C 10 33.33333333333333 LCU 1874 10 33.33333333333333 LCU 6005BE 10 33.33333333333333 LCU 1AD8 10 33.33333333333333 LCU 27C3 10 33.33333333333333 LCU 2780 10 33.33333333333333 LCU 34E4 9 30.0 LCU 2910 9 30.0 LCU 1593 9 30.0 LCU 1A10 9 30.0 LCU 12B0 9 30.0 LCU 602F98 8 26.666666666666668

### Discussão
Essa pergunta é uma consequência do elevado número de alarmes detectados na pergunta 4. Nessa consulta a tentativa era de encontrar casos em que o alarme de Acesa Durante o Dia era permanente e não transitório, gerado por algum mecanismo na LCU. A ideia é que alarmes com alta frequência talvez indiquem que o alarme é permanente.

Entretanto, avaliando essas amostras, percebemos casos de LCUs que passaram vários dias emitindo alarme e simplesmente voltaram a funcionar normalmente depois.
A pergunta a seguir foi criada para detectar esses casos.

## 8. LCUs com dias consecutivos com alarme Acesa Durante o Dia.

Apesar dessa pergunta ser parecida com a anterior essa é muito mais importante para a gerência e o propósito é diferente em duas situações:
1. Os dias com alarme precisam ser consecutivos para a contagem - vou estabelecer mínimo de 3 dias.
2. O alarme deve ter parado em um momento posterior. Vou novamente considerar um período de 2 dias sem alarmes do tipo analisado.

Exemplo: LCU 1000 gera alarme de Aceso Durante o Dia na Data 10/11/2025. Ela deve emitir o alarme também nas datas 11/11/2025 e 12/11/2025. Além disso, o alarme não pode ser emitido nas datas 13/11/2025 e 14/11/2025. O período de dois dias em que o alarme não é emitido é para termos certeza que realmente o alarme parou e o problema não persiste.

A LCU pode ficar mais do que três dias consecutivos com alarme para um resultado ser retornado nessa consulta. No exemplo acima, a LCU 1000 poderia ter iniciado a emitir alarme no dia 01/11/2025, mas deve seguir as regras estabelecidas acima.

Essa pergunta serve para evidenciar à empresa responsável pelo sistema de Telegestão a instabilidade do alarme de Acesa Durante o Dia. As LCUs que apresentam esse alarme ligam a luminária LED durante o dia de forma errada, por vários dias consecutivos e, sem intervenção de equipe de manutenção, a LCU volta a funcionar normal, desligando a lâmpada.
Essa pergunta servirá como evidência em relatório.

*Update. No dia 04/12/2025 foi enviado relatório com as LCUs detectadas por esse trabalho à empresa responsável pela Telegestão. A empresa reconheceu que realmente existe um problema e atualmente estão entrando em contato com o fabricante do equipamento para esclarecimentos.

In [0]:
%sql
CREATE OR REPLACE VIEW viewAcesaDiaConsecutivas AS (
  SELECT v1.*
  FROM viewalarmelcu v1
  WHERE v1.Data <= date_add((SELECT MAX(Data) FROM dimdata), - 4)
  -- v1.Data deve ser tal que garanta existir pelo menos duas datas após os três dias consecutivos, para a clásula AND NOT EXISTS funcionar.
  -- Caso não exista pelo menos duas datas após os três dias avaliados no banco de dados, NOT EXISTS poderá ser avaliado como True indevidamente,
  -- fazendo com que traga como resposta LCU que não deixaram de emitir alarme Acesa Durante o Dia.
  AND v1.NomeAlarme = 'Acesa Durante o Dia'
  AND v1.NomeLCU IN (
    SELECT v2.NomeLCU
    FROM viewalarmelcu v2
    WHERE v1.IDLCU = v2.IDLCU
    AND v2.NomeAlarme = 'Acesa Durante o Dia'
    AND v2.Data < date_add(v1.Data, 3)
    AND v2.Data >= v1.Data
    GROUP BY v2.NomeLCU
    HAVING count(v2.NomeLCU) = 3
  )
  AND NOT EXISTS (
    SELECT 1
    FROM viewalarmelcu v2
    WHERE v1.IDLCU = v2.IDLCU -- Mesma LCU
    AND v1.NomeAlarme = v2.NomeAlarme -- Mesmo Alarme
    AND v2.Data IN (date_add(v1.Data, 3), date_add(v1.Data, 4))
    -- As vezes o sistema de telegestão não comunica um alarme em um dia, por isso o check em dois dias.
  )
)

In [0]:
%sql
SELECT *
FROM viewAcesaDiaConsecutivas
LIMIT 50;

NomeLCU NomeAlarme Data TrimestreContrato IDLCU Latitude Longitude LCU 102C Acesa Durante o Dia 2025-11-24 T16 2543196311596 -11.007403 -37.07467 LCU 10FD Acesa Durante o Dia 2025-11-07 T16 2543196311805 -10.926409 -37.052948 LCU 11F0 Acesa Durante o Dia 2025-11-21 T16 2543196312048 -11.005013 -37.10332 LCU 12B0 Acesa Durante o Dia 2025-11-23 T16 2543196312240 -11.007684 -37.074886 LCU 12BD Acesa Durante o Dia 2025-11-12 T16 2543196312253 -11.004372 -37.072433 LCU 12C3 Acesa Durante o Dia 2025-11-28 T16 2543196312259 -10.927903 -37.04331 LCU 1313 Acesa Durante o Dia 2025-11-27 T16 2543196312339 -11.00379 -37.071953 LCU 151A Acesa Durante o Dia 2025-11-21 T16 2543196312858 -10.993076 -37.062847 LCU 1581 Acesa Durante o Dia 2025-11-27 T16 2543196312961 -10.974787 -37.038902 LCU 1624 Acesa Durante o Dia 2025-11-05 T16 2543196313124 -10.991919 -37.050865 LCU 166E Acesa Durante o Dia 2025-11-11 T16 2543196313198 -10.99065 -37.08284 LCU 166E Acesa Durante o Dia 2025-11-17 T16 2543196313198 -10.99065 -37.08284 LCU 17FD Acesa Durante o Dia 2025-11-16 T16 2543196313597 -10.953673 -37.05541 LCU 1982 Acesa Durante o Dia 2025-11-20 T16 2543196313986 -10.901058 -37.052616 LCU 198D Acesa Durante o Dia 2025-11-23 T16 2543196313997 -10.910608 -37.073624 LCU 19DD Acesa Durante o Dia 2025-11-05 T16 2543196314077 -10.909773 -37.090775 LCU 19FD Acesa Durante o Dia 2025-11-12 T16 2543196314109 -10.911083 -37.090347 LCU 1A10 Acesa Durante o Dia 2025-11-24 T16 2543196314128 -10.997111 -37.0665 LCU 1BB5 Acesa Durante o Dia 2025-11-10 T16 2543196314549 -10.950982 -37.067116 LCU 1F69 Acesa Durante o Dia 2025-11-08 T16 2543196315497 -11.003135 -37.102383 LCU 2068 Acesa Durante o Dia 2025-11-23 T16 2543196315752 -10.9514 -37.06477 LCU 2070 Acesa Durante o Dia 2025-11-22 T16 2543196315760 -10.915651 -37.05037 LCU 2143 Acesa Durante o Dia 2025-11-07 T16 2543196315971 -11.006545 -37.074024 LCU 21EA Acesa Durante o Dia 2025-11-08 T16 2543196316138 -10.923418 -37.08104 LCU 23D9 Acesa Durante o Dia 2025-11-26 T16 2543196316633 -10.926998 -37.04337 LCU 2625 Acesa Durante o Dia 2025-11-25 T16 2543196317221 -10.906286 -37.06331 LCU 2692 Acesa Durante o Dia 2025-11-27 T16 2543196317330 -10.945951 -37.074387 LCU 26CC Acesa Durante o Dia 2025-11-13 T16 2543196317388 -10.92309 -37.081234 LCU 27C3 Acesa Durante o Dia 2025-11-24 T16 2543196317635 -10.990822 -37.080242 LCU 50289B Acesa Durante o Dia 2025-11-06 T16 2543196317851 -10.917454 -37.05744 LCU 290C Acesa Durante o Dia 2025-11-24 T16 2543196317964 -11.002658 -37.071095 LCU 2911 Acesa Durante o Dia 2025-11-06 T16 2543196317969 -11.002638 -37.071095 LCU 5029C2 Acesa Durante o Dia 2025-11-23 T16 2543196318146 -11.104772 -37.129612 LCU 2B00 Acesa Durante o Dia 2025-11-10 T16 2543196318464 -10.99047 -37.049873 LCU 2B79 Acesa Durante o Dia 2025-11-11 T16 2543196318585 -10.991854 -37.062077 LCU 2CD1 Acesa Durante o Dia 2025-11-12 T16 2543196318929 -10.903981 -37.056305 LCU 2DC5 Acesa Durante o Dia 2025-11-08 T16 2543196319173 -10.978833 -37.04162 LCU 2DDC Acesa Durante o Dia 2025-11-23 T16 2543196319196 -10.997847 -37.067177 LCU 2E53 Acesa Durante o Dia 2025-11-12 T16 2543196319315 -11.0006485 -37.069546 LCU 2E93 Acesa Durante o Dia 2025-11-16 T16 2543196319379 -10.99891 -37.068195 LCU 2EF3 Acesa Durante o Dia 2025-11-08 T16 2543196319475 -11.000934 -37.069748 LCU 2FF8 Acesa Durante o Dia 2025-11-22 T16 2543196319736 -10.900838 -37.050343 LCU 3186 Acesa Durante o Dia 2025-11-14 T16 2543196320134 -10.914788 -37.088623 LCU 335B Acesa Durante o Dia 2025-11-17 T16 2543196320603 -11.00063 -37.069553 LCU 34E4 Acesa Durante o Dia 2025-11-19 T16 2543196320996 -10.906165 -37.06475 LCU 503574 Acesa Durante o Dia 2025-11-26 T16 2543196321140 -10.938915 -37.06343 LCU 503963 Acesa Durante o Dia 2025-11-27 T16 2543196322147 -10.957202 -37.069275 LCU 39B0 Acesa Durante o Dia 2025-11-09 T16 2543196322224 -10.931412 -37.0474 LCU 39D7 Acesa Durante o Dia 2025-11-11 T16 2543196322263 -11.013844 -37.0796 LCU 39F2 Acesa Durante o Dia 2025-

### Discussão
Foi possível detectar mais de 100 casos em que a LCU ficou ao menos três dias consecutivos com alarme de acesa durante o dia. Segue exemplo do gráfico de potência medida ao longo do tempo da LCU 10FD:

![LCU 10FD Day Burner.png](./LCU 10FD Day Burner.png)

Onde a seta está apontando é a data 07/11/2025, que é a data retornada na consulta. Essa data representa o primeiro dia de alarme Aceso Durante o Dia considerado na regra criada de três dias consecutivos, conforme explicado no enunciado. Nas datas 08/11 e 09/11 é possível notar que as potências medidas estão acima de 0 e, a partir do dia 10/11, as potência medidas estão em 0.

Gostaria de deixar claro que não existe maneira de consultar isso no sistema de telegestão, a não ser ir entrando em cada LCU individualmente no sistema e checando o gráfico como a figura acima. Esse trabalho foi graças a esse MVP de DW.

### 8.2 LCUs com alarme Acesa Durante durante apenas um dia

A empresa de telegestão sugeriu que não existiam situações em que a LCU emitia alarme de acesa durante o dia por apenas um único dia e depois voltava ao normal. Fiz essa nova consulta para mostrar que existem, sim, essas situações. Inclusive é muito mais comum do que a situação anterior, de dias consecutivos.

In [0]:
%sql
SELECT v1.*
FROM viewalarmelcu v1
WHERE v1.Data < (SELECT MAX(Data) FROM dimdata)
AND v1.NomeAlarme = 'Acesa Durante o Dia'
AND v1.NomeLCU NOT IN (
  SELECT v2.NomeLCU
  FROM viewalarmelcu v2
  WHERE v2.Data IN (date_add(v1.Data, 1), date_add(v1.Data, - 1))
  AND v1.IDLCU = v2.IDLCU
  AND v2.NomeAlarme = V1.NomeAlarme
)
LIMIT 50;

NomeLCU NomeAlarme Data TrimestreContrato IDLCU Latitude Longitude LCU 102C Acesa Durante o Dia 2025-11-15 T16 2543196311596 -11.007403 -37.07467 LCU 1075 Acesa Durante o Dia 2025-11-12 T16 2543196311669 -10.930195 -37.057327 LCU 1181 Acesa Durante o Dia 2025-11-17 T16 2543196311937 -10.935231 -37.08304 LCU 11F4 Acesa Durante o Dia 2025-11-20 T16 2543196312052 -10.976127 -37.066467 LCU 1251 Acesa Durante o Dia 2025-11-12 T16 2543196312145 -10.927792 -37.05294 LCU 12BA Acesa Durante o Dia 2025-11-12 T16 2543196312250 -10.929392 -37.056816 LCU 12DE Acesa Durante o Dia 2025-11-12 T16 2543196312286 -10.929697 -37.05679 LCU 12EE Acesa Durante o Dia 2025-11-17 T16 2543196312302 -10.918912 -37.05501 LCU 1314 Acesa Durante o Dia 2025-11-20 T16 2543196312340 -11.003507 -37.07176 LCU 1337 Acesa Durante o Dia 2025-11-26 T16 2543196312375 -10.93952 -37.080734 LCU 13E6 Acesa Durante o Dia 2025-11-12 T16 2543196312550 -10.94367 -37.053425 LCU 1448 Acesa Durante o Dia 2025-11-12 T16 2543196312648 -10.943247 -37.05486 LCU 1449 Acesa Durante o Dia 2025-11-12 T16 2543196312649 -10.943365 -37.05449 LCU 1484 Acesa Durante o Dia 2025-11-12 T16 2543196312708 -10.938305 -37.06264 LCU 1484 Acesa Durante o Dia 2025-11-14 T16 2543196312708 -10.938305 -37.06264 LCU 15A5 Acesa Durante o Dia 2025-11-20 T16 2543196312997 -10.996395 -37.05423 LCU 17FD Acesa Durante o Dia 2025-11-10 T16 2543196313597 -10.953673 -37.05541 LCU 1869 Acesa Durante o Dia 2025-11-12 T16 2543196313705 -10.93169 -37.058132 LCU 1883 Acesa Durante o Dia 2025-11-08 T16 2543196313731 -11.003507 -37.07175 LCU 18BF Acesa Durante o Dia 2025-11-11 T16 2543196313791 -10.899743 -37.096413 LCU 18CD Acesa Durante o Dia 2025-11-26 T16 2543196313805 -10.937411 -37.08243 LCU 1A5C Acesa Durante o Dia 2025-11-23 T16 2543196314204 -10.929435 -37.05665 LCU 1A84 Acesa Durante o Dia 2025-11-12 T16 2543196314244 -10.9440565 -37.051567 LCU 1AD8 Acesa Durante o Dia 2025-11-16 T16 2543196314328 -10.911613 -37.052845 LCU 1D92 Acesa Durante o Dia 2025-11-12 T16 2543196315026 -10.942734 -37.056057 LCU 1DE4 Acesa Durante o Dia 2025-11-12 T16 2543196315108 -10.929915 -37.056885 LCU 1E36 Acesa Durante o Dia 2025-11-12 T16 2543196315190 -10.931236 -37.057762 LCU 501E85 Acesa Durante o Dia 2025-11-28 T16 2543196315269 -10.94508 -37.070213 LCU 2023 Acesa Durante o Dia 2025-11-18 T16 2543196315683 -10.968747 -37.088406 LCU 220E Acesa Durante o Dia 2025-11-19 T16 2543196316174 -10.92618 -37.043682 LCU 2267 Acesa Durante o Dia 2025-11-12 T16 2543196316263 -10.943308 -37.05433 LCU 2310 Acesa Durante o Dia 2025-11-12 T16 2543196316432 -10.92818 -37.055977 LCU 2323 Acesa Durante o Dia 2025-11-12 T16 2543196316451 -10.9274435 -37.055157 LCU 2409 Acesa Durante o Dia 2025-11-06 T16 2543196316681 -10.995827 -37.065243 LCU 247D Acesa Durante o Dia 2025-11-15 T16 2543196316797 -10.98745 -37.09271 LCU 2607 Acesa Durante o Dia 2025-11-24 T16 2543196317191 -10.994942 -37.09281 LCU 2775 Acesa Durante o Dia 2025-11-12 T16 2543196317557 -10.94388 -37.052746 LCU 27BF Acesa Durante o Dia 2025-11-12 T16 2543196317631 -10.927506 -37.054234 LCU 502946 Acesa Durante o Dia 2025-11-28 T16 2543196318022 -10.94085 -37.065426 LCU 502A07 Acesa Durante o Dia 2025-11-23 T16 2543196318215 -11.097462 -37.141983 LCU 2AA6 Acesa Durante o Dia 2025-11-11 T16 2543196318374 -10.8939085 -37.069824 LCU 2C58 Acesa Durante o Dia 2025-11-12 T16 2543196318808 -10.9284 -37.05594 LCU 2CCE Acesa Durante o Dia 2025-11-12 T16 2543196318926 -10.927647 -37.055363 LCU 2E11 Acesa Durante o Dia 2025-11-12 T16 2543196319249 -10.927814 -37.053413 LCU 2E2A Acesa Durante o Dia 2025-11-28 T16 2543196319274 -10.971907 -37.03676 LCU 2F0C Acesa Durante o Dia 2025-11-12 T16 2543196319500 -10.92772 -37.055473 LCU 2F2C Acesa Durante o Dia 2025-11-12 T16 2543196319532 -10.944155 -37.051655 LCU 2FB2 Acesa Durante o Dia 2025-11-14 T16 2543196319666 -10.925815 -37.05315 LCU 318B Acesa Durante o Dia 2025-11-05 T16 2543196320139 -10.989298 -37.07562 LCU 318B Acesa Durante o Dia 2025-11-1

### Discussão
Quase 400 LCUs ficaram acesas ao menos um dia, mostrando ao time responsável do sistema de telegestão que esse caso existe e é até mais comum do que o anterior.

## 9. Número de LCUs com Alarmes recorrentes.

Alarmes em dias não consecutivos podem indicar que o alarme foi corrigido/supremido em determinada data e o problema retornou em uma data posterior. Importante para determinar se estamos tendo recorrência de defeitos ou se a LCU em questão está sofrendo instabilidade.

A ideia será retornar registros de LCU que possuem determinado alarme na data D e que não possuem esse mesmo alarme na data D + 1 dia. A contagem dessas LCUs retornará quantos defeitos recorrentes ocorrem por LCU, por alarme.
Nessa pergunta, também queremos detectar alarmes originados por instabilidade. Então o critério será D + 1 dia e não será utilizada a heurística de verificar D + 2 dias como em perguntas anteriores.

In [0]:
%sql
CREATE OR REPLACE VIEW viewAlarmesRecorrentes AS (
  SELECT v1.*
  FROM viewalarmelcu v1
  WHERE v1.Data < (SELECT MAX(Data) FROM dimdata)
  AND NOT EXISTS (
    SELECT 1
    FROM viewalarmelcu v2
    WHERE v1.NomeAlarme = v2.NomeAlarme
    AND v1.IDLCU = v2.IDLCU
    AND v2.Data = date_add(v1.Data, 1)
  )
)

In [0]:
%sql
SELECT NomeLCU, NomeAlarme, count(NomeLCU) NumeroOrrencias
FROM viewalarmesrecorrentes
GROUP BY NomeLCU, NomeAlarme
HAVING NumeroOrrencias > 1 -- Maior que 1 para mostrar apenas recorrentes.
ORDER BY NomeLCU, NumeroOrrencias DESC
LIMIT 50;

NomeLCU NomeAlarme NumeroOrrencias LCU 101F Apagada Durante a Noite 4 LCU 102C Acesa Durante o Dia 3 LCU 102C Sem Comunicação 2 LCU 108A Acesa Durante o Dia 2 LCU 1265 Sem Comunicação 2 LCU 1294 Acesa Durante o Dia 2 LCU 12B2 Tensão Baixa 3 LCU 12BD Acesa Durante o Dia 2 LCU 1375 Acesa Durante o Dia 2 LCU 1405 Acesa Durante o Dia 2 LCU 1484 Acesa Durante o Dia 2 LCU 1585 Sem Comunicação 2 LCU 166E Acesa Durante o Dia 2 LCU 177A Acesa Durante o Dia 5 LCU 17A3 Sem Comunicação 2 LCU 17EA Acesa Durante o Dia 2 LCU 17FD Acesa Durante o Dia 2 LCU 1849 Acesa Durante o Dia 2 LCU 189C Acesa Durante o Dia 2 LCU 198E Acesa Durante o Dia 2 LCU 1A10 Acesa Durante o Dia 2 LCU 1AD8 Acesa Durante o Dia 2 LCU 1C01 Sem Comunicação 2 LCU 1CFD Acesa Durante o Dia 5 LCU 1D07 Sem Comunicação 2 LCU 1D19 Sem Comunicação 2 LCU 1D46 Acesa Durante o Dia 4 LCU 1F69 Acesa Durante o Dia 2 LCU 1F81 Falha no Driver 6 LCU 2068 Acesa Durante o Dia 2 LCU 20C4 Sem Comunicação 2 LCU 2114 Sem Comunicação 2 LCU 2143 Acesa Durante o Dia 3 LCU 21C0 Sem Comunicação 2 LCU 220B Sem Comunicação 2 LCU 220E Acesa Durante o Dia 2 LCU 220F Sem Comunicação 2 LCU 23DB Sem Comunicação 2 LCU 2409 Acesa Durante o Dia 2 LCU 2457 Falha no Driver 8 LCU 248D Sem Comunicação 2 LCU 24BB Sem Comunicação 2 LCU 2607 Acesa Durante o Dia 2 LCU 2625 Acesa Durante o Dia 2 LCU 2780 Acesa Durante o Dia 2 LCU 287A Sem Comunicação 2 LCU 28A5 Sem Comunicação 2 LCU 28D2 Sem Comunicação 2 LCU 290C Acesa Durante o Dia 3 LCU 2910 Acesa Durante o Dia 2

In [0]:
%sql
SELECT NomeLCU, count(NomeLCU) NumeroOrrencias
FROM viewalarmesrecorrentes
WHERE NomeAlarme = 'Sem Comunicação'
GROUP BY NomeLCU
HAVING NumeroOrrencias > 1
ORDER BY NumeroOrrencias DESC
LIMIT 50;

NomeLCU NumeroOrrencias LCU 604588 5 LCU 60136F 5 LCU 6012C4 4 LCU 6012A1 3 LCU 3768 3 LCU 5039ED 3 LCU 3769 3 LCU 600A14 3 LCU 604578 3 LCU 604B65 3 LCU 50211D 3 LCU 6009FE 3 LCU 600CA9 3 LCU 6051F8 3 LCU 6042BB 3 LCU 503969 3 LCU 601490 3 LCU 6039BF 3 LCU 603728 3 LCU 604E94 3 LCU 3A2F 2 LCU 287A 2 LCU 33BE 2 LCU 2A20 2 LCU 60353A 2 LCU 32CB 2 LCU 2B00 2 LCU 2A55 2 LCU 600186 2 LCU 1265 2 LCU 35FB 2 LCU 600710 2 LCU 601560 2 LCU 604C05 2 LCU 603D3A 2 LCU 60120C 2 LCU 60281A 2 LCU 605600 2 LCU 601833 2 LCU 603EFE 2 LCU 248D 2 LCU 2A4A 2 LCU 50343B 2 LCU 29CE 2 LCU 32BD 2 LCU 3778 2 LCU 5025AE 2 LCU 602AEC 2 LCU 3319 2 LCU 6038FF 2

### Discussão

O caso de alarmes recorrentes é mais útil quando analisamos o número de ocorrências de alarmes Sem Comunicação. Foi mencionado anteriormente que na maioria dos casos esse alarmes é consequência de uma falta de energia que precisa ser reestabelecida pela equipe de manutenção. Pórem, existem casos em que esse alarme surge devido a falhas no sinal da rede LoraWan, que na verdade é o caso em que faz mais sentido com o nome do alarme.

Esses casos de falha no sinal da rede podem ser verificados com essa consulta de recorrências em um intervalo curto de tempo. As LCUs 60136F, 604588 e 6012C4, por exemplo, são praticamente vizinhas umas das outras, indicando que talvez exista uma falha na cobertura nesse local.

## 10. Alarmes por bairro no trimestre do contrato.

In [0]:
%sql
CREATE OR REPLACE VIEW viewConecta AS (
  SELECT IDLCU, NomeLCU, dimlcu.Latitude, dimlcu.Longitude, Data, TrimestreContrato, IDPontoServico, Bairro, MarcoContrato, NomeAlarme
  FROM fatoalarme
  LEFT JOIN dimtipoalarme ON fatoalarme.AlarmeKey = dimtipoalarme.AlarmeKey
  LEFT JOIN dimpontoservico ON fatoalarme.PontoServicoKey = dimpontoservico.PontoServicoKey
  LEFT JOIN dimdata ON fatoalarme.DataKey = dimdata.DataKey
  LEFT JOIN dimlcu ON fatoalarme.LCUKey = dimlcu.LCUKey
);

In [0]:
%sql
SELECT Bairro, NomeAlarme, count(DISTINCT IDLCU) NumeroAlarmes
FROM viewConecta
WHERE TrimestreContrato IN (
  SELECT TrimestreContrato
  FROM viewConecta
  WHERE Data = (SELECT MAX(Data) FROM dimdata)
)
GROUP BY Bairro, NomeAlarme
ORDER BY Bairro, NumeroAlarmes DESC;

Bairro NomeAlarme NumeroAlarmes 13 de julho Sem Comunicação 16 13 de julho Acesa Durante o Dia 3 13 de julho Falha no Driver 2 17 de março Acesa Durante o Dia 3 17 de março Sem Comunicação 2 17 de março Falha no Driver 1 18 do forte Tensão Baixa 1 18 do forte Falha no Driver 1 Aeroporto Sem Comunicação 14 Aeroporto Acesa Durante o Dia 8 Aeroporto Falha no Driver 4 Aeroporto Apagada Durante a Noite 1 América Sem Comunicação 7 Aruana Acesa Durante o Dia 23 Aruana Sem Comunicação 9 Aruana Falha no Driver 7 Aruana Apagada Durante a Noite 1 Atalaia Sem Comunicação 27 Atalaia Acesa Durante o Dia 19 Atalaia Falha no Driver 5 Atalaia Apagada Durante a Noite 2 Atalaia Tensão Baixa 1 Capucho Sem Comunicação 2 Capucho Acesa Durante o Dia 1 Centro Sem Comunicação 15 Centro Acesa Durante o Dia 6 Centro Falha no Driver 4 Cidade nova Acesa Durante o Dia 1 Cirurgia Falha no Driver 1 Cirurgia Acesa Durante o Dia 1 Coroa do meio Sem Comunicação 52 Coroa do meio Falha no Driver 7 Coroa do meio Acesa Durante o Dia 2 Coroa do meio Apagada Durante a Noite 2 Dom luciano Acesa Durante o Dia 2 Farolândia Sem Comunicação 8 Farolândia Falha no Driver 4 Farolândia Acesa Durante o Dia 4 Farolândia Apagada Durante a Noite 2 Gameleira Sem Comunicação 8 Gameleira Falha no Driver 3 Gameleira Acesa Durante o Dia 2 Getúlio vargas Acesa Durante o Dia 3 Getúlio vargas Sem Comunicação 1 Getúlio vargas Falha no Driver 1 Grageru Sem Comunicação 46 Grageru Acesa Durante o Dia 36 Grageru Apagada Durante a Noite 30 Grageru Falha no Driver 4 Industrial Acesa Durante o Dia 10 Industrial Sem Comunicação 9 Industrial Tensão Baixa 2 Industrial Tensão Alta 1 Inácio barbosa Sem Comunicação 6 Inácio barbosa Acesa Durante o Dia 4 Inácio barbosa Tensão Alta 4 Inácio barbosa Falha no Driver 2 Jabotiana Falha no Driver 2 Jabotiana Tensão Baixa 2 Jabotiana Acesa Durante o Dia 1 Japãozinho Sem Comunicação 1 Jardins Sem Comunicação 12 Jardins Apagada Durante a Noite 4 Jardins Tensão Alta 4 Jardins Falha no Driver 3 Jardins Tensão Baixa 1 José conrado de araújo Sem Comunicação 3 José conrado de araújo Falha no Driver 1 Lamarão Sem Comunicação 1 Luzia Acesa Durante o Dia 36 Luzia Sem Comunicação 24 Luzia Apagada Durante a Noite 24 Luzia Falha no Driver 1 Luzia Tensão Alta 1 Marivan Sem Comunicação 1 Mosqueiro Sem Comunicação 6 Mosqueiro Acesa Durante o Dia 3 Mosqueiro Falha no Driver 2 Novo paraíso Sem Comunicação 3 Novo paraíso Acesa Durante o Dia 1 Olaria Sem Comunicação 1 Olaria Falha no Driver 1 Palestina Acesa Durante o Dia 1 Ponto novo Falha no Driver 3 Ponto novo Apagada Durante a Noite 2 Ponto novo Acesa Durante o Dia 2 Ponto novo Sem Comunicação 1 Porto d'antas Sem Comunicação 3 Porto d'antas Apagada Durante a Noite 1 Robalo Sem Comunicação 8 Robalo Acesa Durante o Dia 2 Robalo Falha no Driver 1 Salgado filho Acesa Durante o Dia 5 Salgado filho Apagada Durante a Noite 1 Salgado filho Falha no Driver 1 Salgado filho Tensão Baixa 1 Salgado filho Sem Comunicação 1 Santa maria Sem Comunicação 1 Santo antônio Acesa Durante o Dia 5 Santo antônio Sem Comunicação 1 Santos dumont Sem Comunicação 2 Santos dumont Acesa Durante o Dia 1 Santos dumont Apagada Durante a Noite 1 Santos dumont Falha no Driver 1 Siqueira campos Sem Comunicação 5 Siqueira campos Acesa Durante o Dia 1 Siqueira campos Apagada Durante a Noite 1 Siqueira campos Falha no Driver 1 Soledade Acesa Durante o Dia 3 Suíça Sem Comunicação 5 Suíça Falha no Driver 2 São conrado Sem Comunicação 15 São conrado Acesa Durante o Dia 1 São josé Sem Comunicação 3 São josé Falha no Driver 2 São josé Acesa Durante o Dia 1 São josé dos náufragos Sem Comunicação 5

### Discussão
Para finalizar, uma consulta para mostrar quantas LCUs distintas por bairro no trimestre do contrato emitiram alarme. Essa consulta é útil para tentar identificar uma região que seja mais afetada por algum tipo de alarme.